data_model/src/validator.rs

```diff
@@ -79,6 +79,12 @@ pub mod model {
 /// Result type that every validator should return.
 pub type Result<T = (), E = crate::ValidationFail> = core::result::Result<T, E>;
 
+/// Migration error type.
+pub type MigrationError = String;
+
+/// Result type for `migration()` entrypoint.
```

Comment:
`migration()`? Or `migrate()`?

Since there is no way to make a doc link to the exact function name, maybe omit the exact name here at all?

```suggestion
/// Result type for a validator's `migrate()` entrypoint.
```